### DISCUSSION TITLE
Help to solve an error with the Raspberry Pi Pico 2 board

### DISCUSSION THREAD

**Discussion by Motokegor** [earlephilhower/arduino-pico#2626] [General]
![5cdbae87-9f9f-41c1-aa4f-5be3fea9531a](https://github.com/user-attachments/assets/6d13789f-58d8-48d7-9f1a-2f2ac8e7c15c)
![9610a3fa-57db-4886-bbad-c1ed8b5b2d38](https://github.com/user-attachments/assets/697cd478-ab07-4803-bbb1-32fa9d9cd5a9)
![f082b3e8-649c-46b0-b5a2-4dc2dd850eff](https://github.com/user-attachments/assets/bdc48d98-70fa-4f93-acc7-aee932c7c80c)
I'm trying to upload a sketch as described here https://github.com/earlephilhower/arduino-pico?tab=readme-ov-file#uploading-sketches but it's not clear whether it works or not. I press the Start Debugging button and the error shown on the screenshot pops up.

**Comment by maxgerhardt** [comment]
"unable to find matching CMSIS device" is telling you that OpenOCD can't find a debugger probe.

For debugging to work, you **need** to have an external debugging probe attached to the SWD pins of your Pico 2 board. One such way is to take a second Pico (RP2040) or "DebugProbe" and load the https://github.com/raspberrypi/debugprobe/releases firmware firmware on it (e.g., ` debugprobe_on_pico.uf2 `), and then connect that debug probe according to https://datasheets.raspberrypi.com/pico/getting-started-with-pico.pdf, Appendix A.
